### PR TITLE
Code clean-up push to fix some of the 2.0 deprecation warnings in the application

### PR DESCRIPTION
**Pull request by rauhryan**
Implements so good old fashioned cleanup and grunt work to fix several outstanding deprecations and cleanup the console experience while developing on HuBoard day to day

### CODE REVIEW

**Comment by dahlbyk** [comment]
Should I expect zero Ember deprecation warnings under this PR? I'm still seeing several instances of:

> Using the '{{#each item in model}}' form of the {{#each}} helper is deprecated. Please use the block param form instead ('{{#each model as |item|}}'). See 
http://emberjs.com/guides/deprecations/#toc_code-in-code-syntax-for-code-each-code for more details.

**Comment by rauhryan** [comment]
@dahlbyk are you getting those warnings in the browser console? I am not getting them

**Comment by dahlbyk** [comment]
> are you getting those warnings in the browser console? I am not getting them

On build. Those fixes look good.

File: ember-app/app/controllers/settings/links/index.js [hunk 1]
```diff
@@ -2,13 +2,13 @@ import Ember from 'ember';
 import { debouncedObserver } from 'app/utilities/observers';
 
 var SettingsLinksIndexController = Ember.Controller.extend({
-  needs: ['application', 'settings/links'],
```
Comment (discorick) [inline ember-app/app/controllers/settings/links/index.js]: Does this controller not actually need `application` controller injection?

**Comment by discorick** [comment]
The date picker on Milestone columns seems broken, attempting to assign a date no loner updates the date field.

File: ember-app/app/controllers/settings/links/index.js (hunk 1 above)
Comment (rauhryan) [inline ember-app/app/controllers/settings/links/index.js]: Yes, line 11 references a property on it.

**Comment by rauhryan** [comment]
Nice catch on the milestones date picker @discorick 

I didn't even think to check it. 

I just pushed a fix.

**Pull request merged**